### PR TITLE
pypy: enable aarch64-linux platform

### LINKED ISSUE #104123: Pypy should support aarch64-linux, but doesn't
**Describe the bug**
Both `pypy3` and `pypy2` should, as of 7.1.1(https://morepypy.blogspot.com/2019/07/pypy-jit-for-aarch64.html) support the `aarch64-linux` platform.

We, however, do not have that platform enabled.

**Notify maintainers**

@andersk 
Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: pypy3, pypy2
```

### PR DESCRIPTION
**Pull request by lovesegfault**
###### Motivation for this change

Closes #104123

~Still testing.~ It works!

cc. @andersk

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by andersk — APPROVED** [review]

File: pkgs/development/interpreters/python/pypy/default.nix [hunk 1]
```diff
@@ -151,7 +151,7 @@ in with passthru; stdenv.mkDerivation rec {
     homepage = "http://pypy.org/";
     description = "Fast, compliant alternative implementation of the Python language (${pythonVersion})";
     license = licenses.mit;
-    platforms = [ "i686-linux" "x86_64-linux" "x86_64-darwin" ];
+    platforms = [ "aarch64-linux" "i686-linux" "x86_64-linux" "x86_64-darwin" ];
```
Comment (prusnak) [inline pkgs/development/interpreters/python/pypy/default.nix:154]: Shouldn't we just use `platforms = platforms.unix;` at this point?

Comment (lovesegfault) [reply pkgs/development/interpreters/python/pypy/default.nix:154]: Maybe? Does it work on `ppc`?

Comment (lovesegfault) [reply pkgs/development/interpreters/python/pypy/default.nix:154]: A quick google leads me to believe "probably no"

Comment (lovesegfault) [reply pkgs/development/interpreters/python/pypy/default.nix:154]: My point being that pypy _doesn't_ just support posix OS', but relies heavily on the underlying ISA

Comment (prusnak) [reply pkgs/development/interpreters/python/pypy/default.nix:154]: Sounds right. Thanks for explanation.

**Pull request merged**